src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCheck.java

```diff
@@ -228,6 +229,23 @@ public void visitToken(DetailAST ast) {
         }
     }
 
+    /**
+     * Converts the Unicode code point at index {@code stringArrayIndex} to it's UTF-16
+     * representation, then checks if the character is whitespace. Note that the given
+     * index {@code stringArrayIndex} should correspond to the location of the character
+     * to check in the string, not in code points.
+     *
+     * @param codePoints the array of Unicode code points
+     * @param stringArrayIndex the index of the character to check
+     * @return true if character at {@code stringArrayIndex} is whitespace
+     */
+    private static boolean isCodePointWhitespace(int[] codePoints, int stringArrayIndex) {
```

Comment:
I had considered adding this method to `CheckUtil`, but so far there is only one other possible usage in another check (`WhitespaceAroundCheck`).

Reply:
May be there are same issues with other whitespace checks, e.g. GenericWhitespace, NoWhitespaceBeforeCaseDefaultColon, WhitespaceAfter?

Reply:
Issue opened for further investigation at https://github.com/checkstyle/checkstyle/issues/10924

Reply:
issue is approved, lets proceed to review fix for this Check only.

Reply:
@nmancus1 , please rename `stringArrayIndex` to `index`, as it is index in codepoints array.
name method `isWhitespace` by argument it is clear that it is for codepoint

Reply:
Done